gax-grpc/src/main/java/com/google/api/gax/grpc/GrpcClientCalls.java

```diff
@@ -90,4 +99,101 @@
       return channel.newCall(descriptor, callOptions);
     }
   }
+
+  /**
+   * A work-alike of {@link io.grpc.stub.ClientCalls#futureUnaryCall(ClientCall, Object)}.
+   *
+   * <p>The only difference is that unlike grpc-stub's implementation. This implementation doesn't
+   * wait for trailers to resolve a unary RPC. This can save milliseconds when the server is
+   * overloaded.
+   */
+  @BetaApi
+  static <RequestT, ResponseT> ApiFuture<ResponseT> eagerFutureUnaryCall(
+      ClientCall<RequestT, ResponseT> clientCall, RequestT request) {
+    // Start the call
+    GrpcFuture<ResponseT> future = new GrpcFuture<>(clientCall);
+    clientCall.start(new EagerFutureListener<>(future), new Metadata());
+
+    // Send the request
+    try {
+      clientCall.sendMessage(request);
+      clientCall.halfClose();
+      // Request an extra message to detect misconfigured servers
```

Comment:
I wanted to minimize the behavioral differences and its fairly cheap. If you feel strongly, I can remove it. let me know

Reply:
Behavioral differences between what and what? I would still recommend removing it though.

Reply:
Difference between grpc behavior and this. It will also ensure that the call is properly closed as opposed to hanging indefinitely if a second message is sent. Overall I think it makes debugging easier